src/stdlib_io.fypp

```diff
@@ -122,7 +127,12 @@ contains
       #:endif
 
       ! determine number or rows
-      nrow = number_of_rows(s)
+      nrow = number_of_rows(s) - skiprows_
+
+      if ( nrow < 0 ) call error_stop("loadtxt: skipping more rows than present.")
```

Comment:
Should this really be a fatal error?

Reply:
We could also cut of `nrow` at 0, which would result in an empty array.

Reply:
This would be the `numpy` behavior. So I would also be in favor of allowing negative values for `skiprows`.

Reply:
gfortran also crashes when trying to read after EOF. So having a meaning full error message is preferable IMO.

Reply:
Why not using `iostat` and `iomsg` instead of this check, and returning an appropriate message or an zero-size array when appropriate?

Reply:
Unfortunately, I am not familiar with `iostat` and `iomsg`. This would only affect the following loop, right?
```fortran
do i = 1, skiprows_
  read(s, *)
end do
```